editor/plugins/script_editor_plugin.cpp

```diff
@@ -1347,34 +1348,35 @@ void ScriptEditor::_menu_option(int p_option) {
 				scr->reload(true);
 
 			} break;
+
 			case FILE_RUN: {
 				Ref<Script> scr = current->get_edited_resource();
 				if (scr == nullptr || scr.is_null()) {
-					EditorNode::get_singleton()->show_warning(TTR("Can't obtain the script for running."));
+					EditorToaster::get_singleton()->popup_str(TTR("Cannot run the edited file because it's not a script."), EditorToaster::SEVERITY_WARNING);
 					break;
 				}
-				if (!scr->is_tool()) {
-					EditorNode::get_singleton()->show_warning(TTR("Script is not in tool mode, will not be able to run."));
-					return;
-				}
 
 				current->apply_code();
-				Error err = scr->reload(false); //hard reload script before running always
 
-				if (err != OK) {
-					EditorNode::get_singleton()->show_warning(TTR("Script failed reloading, check console for errors."));
+				Error err = scr->reload(false); // Always hard reload the script before running.
+				if (err != OK || !scr->is_valid()) {
+					EditorToaster::get_singleton()->popup_str(TTR("Cannot run the script because it contains errors, check the output log."), EditorToaster::SEVERITY_WARNING);
+					return;
+				}
+				if (!scr->is_tool()) {
+					EditorToaster::get_singleton()->popup_str(TTR("Cannot run the script because it's not a tool script (add the @tool annotation at the top)."), EditorToaster::SEVERITY_WARNING);
```

Comment:
```suggestion
					String error_msg = TTR("Cannot run the script because it's not a tool script.");
					if (scr->get_class() == "GDScript") {
						error_msg = TTR("Cannot run the script because it's not a tool script (add the @tool annotation at the top).");
					}
					EditorToaster::get_singleton()->popup_str(error_msg, EditorToaster::SEVERITY_WARNING);
```

Reply:
I understand why you'd suggest that and I thought about it when adjusting the message. I could've sworn we already had a similar situation with one of the errors somewhere else, but I can't find it right now. I considered that even a GDScript-specific hint would be better than no hint (I'm also not really sure if you can run other scripts from the editor).

It's probably a bigger discussion worth having but for now I can make the change you suggest.